stino-x/Blog-Rails-
# [Feature/Test] Implement Model Validations and Tests

This pull request introduces comprehensive model validations and associated tests for the User, Post, Comment, and Like models. The primary focus is on ensuring data integrity, model associations, and the correct functioning of callbacks.

Changes Made
User Model:

Implemented validations for the presence of the name attribute.
Added tests for the recent_posts custom method.
Post Model:

Included validations for title presence, title length, and numericality of counters.
Implemented a callback to update the user's posts_counter after creating a post.
Added tests for validations, callbacks, and custom methods like recent_comments.
Comment Model:

Introduced a validation for the presence of the text attribute.
Implemented tests for the comment's association with a post, comments counter update, and text presence validation.
Like Model:

Added validations for the presence of user and post associations.
Implemented tests for the update_likes_counter callback.
How to Test
Run RSpec tests for each model using the following commands:
bash
Copy code
rspec spec/models/user_spec.rb
rspec spec/models/post_spec.rb
rspec spec/models/comment_spec.rb
rspec spec/models/like_spec.rb
Additional Notes
Ensure that all tests pass successfully.
If any issues are identified, please review and address them accordingly.
This pull request aims to enhance the robustness of our application by implementing thorough model validations and tests. Your feedback and reviews are highly appreciated.
